### PR TITLE
Improvements to UI-related documentation

### PR DESCRIPTION
**Pull request by florian-h05**
- Persistence config: Mention UI support & Link to UI docs
- Add links to configuration/tutorial docs from Main UI pages
- Minor wording improvements to UI docs
- Building pages: Add CSS section
- Expand sidebar for Main UI devtools & Creating Personal Widgets
- Add images for widget developer, model and persistence settings page.

### CODE REVIEW

File: mainui/settings/rules.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ Rules can be as simple as a directive to turn on a single light at a given time
 
 Community made Rule Templates can also be added to your system from the **Automation** tab of the add-on store.
 
+There is a [rules concepts page]({{base}}/concepts/rules.html) that explains the idea of rules in detail, and a comprehensive introduction to rules in the [rules tutorial]({{base}}/tutorial/rules_basic.html).
```
Comment (stefan-hoehn) [inline mainui/settings/rules.md:14]: missing . at the end.

File: ui/building-pages.md [hunk 2]
```diff
@@ -94,19 +94,19 @@ No matter which type of page you’re editing, the designer will feature black b
...
+- Remove the widget
 
 Configuring the widget with the config sheet is of course more user-friendly than editing the YAML, as the options are usually organized logically in groups and feature a description:
 
 ![Configuring a Widget](./images/widget_config_sheet.png)
 
 However, it's important to know that there are limitations and sometimes editing the YAML directly will be best, because:
 
-1. not all options are described, since widgets are often wrappers for a similar concept in the library it's based on, either Framework7, ECharts, Leaflet, or other specialized libraries.
+1. Not all options are described, since widgets are often wrappers for a similar concept in the library it's based on, either Framework7, ECharts, Leaflet, or other specialized libraries.
```
Comment (stefan-hoehn) [inline ui/building-pages.md:109]: I would rather use "it is" in writing rather than "it's"

File: ui/building-pages.md [hunk 3]
```diff
@@ -147,6 +147,19 @@ To dynamically configure components based on data changed on runtime, expression
 
 Please refer to [widget expressions](widget-expressions-variables.html) for more information.
 
+## Modifying Styling with CSS
```
Comment (stefan-hoehn) [inline ui/building-pages.md:150]: Very helpful addition <3

**Review by stefan-hoehn — CHANGES_REQUESTED** [review]
Very small review request.

Great you used two more collapsed menus that may even grow more in the future as it keeps the main menu smaller.

File: mainui/settings/rules.md (hunk 1 above)
Comment (florian-h05) [reply mainui/settings/rules.md:14]: Hmm, I think the dot is there.

**Review by stefan-hoehn — APPROVED** [review]

**Pull request merged**